jobs/JGLOBAL_STAGE_IC

```diff
@@ -5,13 +5,99 @@ source "${HOMEgfs}/ush/jjob_header.sh" -e "stage_ic" -c "base stage_ic"
 
 # Restart conditions for GFS cycle come from GDAS
 # shellcheck disable=SC2153
-rCDUMP=${RUN}
+rRUN=${RUN}
 # shellcheck disable=SC2153
-[[ ${RUN} = "gfs" ]] && export rCDUMP="gdas"
-export rCDUMP
+[[ ${RUN} = "gfs" ]] && export rRUN="gdas"
 
-# Execute the Script
-"${SCRgfs}/exglobal_stage_ic.sh"
+# Define significant cycles
+# Locally scoped variables and functions
+# shellcheck disable=SC2153
+half_window=$(( assim_freq / 2 ))
+current_cycle="${PDY}${cyc}"
+previous_cycle=$(date --utc -d "${current_cycle:0:8} ${current_cycle:8:2} - ${assim_freq} hours" +%Y%m%d%H)
+current_cycle_begin=$(date --utc -d "${current_cycle:0:8} ${current_cycle:8:2} - ${half_window} hours" +%Y%m%d%H)
+current_cycle_end=$(date --utc -d "${current_cycle:0:8} ${current_cycle:8:2} + ${half_window} hours" +%Y%m%d%H)
+current_cycle_offset=$(date --utc -d "${PDY} ${cyc} + ${OFFSET_START_HOUR} hours" +%Y%m%d%H)
+
+# Define model start date for current_cycle as the time the forecast will start
+if [[ "${DOIAU:-NO}" == "YES" ]]; then
+  model_start_date_current_cycle="${current_cycle_begin}"
+else
+  if [[ "${REPLAY_ICS:-NO}" == "YES" ]]; then
+    model_start_date_current_cycle=${current_cycle_end}
+  else
+    model_start_date_current_cycle=${current_cycle}
+  fi
+fi
+export previous_cycle current_cycle_offset model_start_date_current_cycle
+
+# Define MEMDIR_ARRAY
+MEMDIR_ARRAY=()
+if [[ "${RUN:-}" = "enkfgdas" || "${RUN:-}" = "gefs" ]]; then
+  if [[ "${RUN:-}" = "gefs" ]]; then
+    ii_start=0
+  elif [[ "${RUN:-}" = "enkfgdas" ]]; then
+    ii_start=1
+  fi
+  # Populate the member_dirs array based on the value of NMEM_ENS
+  for ((ii = "${ii_start}"; ii <= "${NMEM_ENS:-0}"; ii++)); do
+    MEMDIR_ARRAY+=("mem$(printf "%03d" "${ii}")")
+  done
+else
+  MEMDIR_ARRAY+=("")
+fi
+
+# Initialize return code
+err=0
+
+###############################################################
+for MEMDIR in "${MEMDIR_ARRAY[@]}"; do
+
+  # Export MEMDIR; need even if empty
+  export MEMDIR
```

Comment:
I committed the `MEMDIR=${MEMDIR}` update to my branch @ https://github.com/NOAA-EMC/global-workflow/pull/2651/commits/55e47c67cd26c1b12f65d5278a09b851c63d8d78

Reply:
> Thoughts on why `MEMDIR` doesn't get into the declare command for the enkfgdas job but did for the gefs job?

MEMDIR is marked for export in the gefs `config.base` (but not the gfs one). While we tend to re-export every time we change a variable, "exported" is really just a condition that attaches to the variable, even if it changes.

The variable has to be exported because `envsubst` (which `declare_from_tmpl()` uses under the hood) only considers exported variables (I forgot this).

Reply:
> The variable has to be exported because envsubst (which declare_from_tmpl() uses under the hood) only considers exported variables (I forgot this).

Good to know!